internal/impl/mysql/snapshot.go

```diff
@@ -0,0 +1,216 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Licensed as a Redpanda Enterprise file under the Redpanda Community
+// License (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+// https://github.com/redpanda-data/connect/v4/blob/main/licenses/rcl.md
+
+package mysql
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strings"
+
+	"github.com/go-mysql-org/go-mysql/mysql"
+	"github.com/redpanda-data/benthos/v4/public/service"
+)
+
+// Snapshot represents a structure that prepares a transaction
+// and creates mysql consistent snapshot inside the transaction
+type Snapshot struct {
+	db *sql.DB
+	tx *sql.Tx
+
+	lockConn     *sql.Conn
+	snapshotConn *sql.Conn
+
+	logger *service.Logger
+	ctx    context.Context
+}
+
+// NewSnapshot creates new snapshot instance
+func NewSnapshot(ctx context.Context, logger *service.Logger, db *sql.DB) *Snapshot {
+	return &Snapshot{
+		db:     db,
+		ctx:    ctx,
+		logger: logger,
+	}
+}
+
+func (s *Snapshot) prepareSnapshot(ctx context.Context) (*mysql.Position, error) {
+	var err error
+	// Create a separate connection for FTWRL
+	s.lockConn, err = s.db.Conn(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create lock connection: %v", err)
+	}
+
+	// Create another connection for the snapshot
+	s.snapshotConn, err = s.db.Conn(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create snapshot connection: %v", err)
+	}
+
+	// 1. Start a consistent snapshot transaction
+	s.tx, err = s.snapshotConn.BeginTx(ctx, &sql.TxOptions{
+		Isolation: sql.LevelRepeatableRead,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to start transaction: %v", err)
+	}
+
+	// Execute START TRANSACTION WITH CONSISTENT SNAPSHOT
```

Comment:
This is not a helpful comment. Maybe better would be to explain why we need to use this transaction option?